src/analog/m2kanalogin_impl.cpp

```diff
@@ -62,14 +49,15 @@ M2kAnalogInImpl::M2kAnalogInImpl(iio_context * ctx, std::string adc_dev, bool sy
 	m_filter_compensation_table[1E4] = 1.20;
 	m_filter_compensation_table[1E3] = 1.26;
 
-	m_samplerate = 1E8;
+	m_samplerate = m_m2k_adc->getDoubleValue("sampling_frequency");
```

Comment:
Shouldn't we do this only in syncDevice() (so if we choose not to sync it, we just have the default values)?

Reply:
Right, my mistake.